samples/capture-to-file/capture-to-file.html

```diff
@@ -1,133 +1,96 @@
 <!DOCTYPE html>
-<html>
 
-<head>
-  <meta charset="UTF-8">
-  <meta http-equiv="origin-trial" content="ArdlZia9G23wi6S2x/vVoTla5x9r1wtreSPqhUq36tpGH7HRmAkPgpBnpkfePFeClDJDzzYAhtDPoI5hJioArAYAAABjeyJvcmlnaW4iOiJodHRwczovL3czYy5naXRodWIuaW86NDQzIiwiZmVhdHVyZSI6IldlYkNvZGVjcyIsImV4cGlyeSI6MTYzODQwMzE5OSwiaXNTdWJkb21haW4iOnRydWV9">
-  <title>WebCodecs API demo: Encoding and Decoding</title>
-  <style>
-    button {
-      background-color: #555555;
-      border: 2px;
-      border-radius: 2px;
-      color: white;
-      padding: 15px 32px;
-      width: 1280px;
-      text-align: center;
-      display: block;
-      font-size: 16px;
-    }
-  </style>
-</head>
+<meta charset="UTF-8">
+<title>WebCodecs API demo: Encoding and Decoding</title>
+<style>
+  button {
+    background-color: #555555;
+    border: 2px;
+    border-radius: 2px;
+    color: white;
+    padding: 15px 32px;
+    width: 1280px;
+    text-align: center;
+    display: block;
+    font-size: 16px;
+  }
+</style>
 
-<body>
-  <button id="record" width=1280 onclick="saveFile()">Record</button>
-  <video id ="src" autoplay muted width=1280 height=720></video>
-  <script type="text/javascript" src="webm-writer2.js"></script>
-  <script>
-    var writeable;
-    var videoWriter;
-    var stop = false;
-    let codec_string = "vp09.00.10.08";
-    var video;
-    var stream;
+<button id="record" width=1280 onclick="onButtonClicked()">Record</button>
+<video id ="src" autoplay muted width=1280 height=720></video>
+<script>
+    let video = document.getElementById('src');
+    let button = document.getElementById('record');
+    let encodeWorker = null;
+    let stream = null;
+    let videoTrack = null;
+
+    async function startRecording() {
+      console.assert(button.innerText == 'Record');
+      button.disabled = true;
```

Comment:
I think (but I'm not sure) that disabling the button at the top of the function, and re-enabling it at the bottom is unnecessary, since the style won't be recomputed during that time (e.g. you can do/undo style changes as many times as you want between window.rAF calls, they only will be visible after rAF).

Reply:
In this case the await on l.32 gives the style a chance to be computed. I added this because I found the save-file-picker didn't always get focus when it opened and I was able to press "start" multiple times, putting the code into a weird state.